src/tbb_concurrent_queue_scheduler.cpp

```diff
@@ -231,7 +233,7 @@ namespace CnC {
 					m_queues[_aff].push( stepInstance );
 				}
             } else {
-                m_gQueue->push( stepInstance );

```

Comment:
Looks like you understood my question just fine. I didn't understand how the wakeups work. Thanks for explaining.